lib/matplotlib/tests/test_axes.py

```diff
@@ -3289,6 +3289,144 @@ def test_hist_step_bottom():
     ax.hist(d1, bottom=np.arange(10), histtype="stepfilled")
 
 
+def test_hist_stepfilled_geometry():
+    bins = [0, 1, 2, 3]
+    data = [0, 0, 1, 1, 1, 2]
+    _, _, (polygon, ) = plt.hist(data,
+                                 bins=bins,
+                                 histtype='stepfilled')
+
+    xy = [[0, 0], [0, 2], [1, 2], [1, 3], [2, 3], [2, 1], [3, 1],
+                   [3, 0], [2, 0], [2, 0], [1, 0], [1, 0], [0, 0]]
+
+    assert (polygon.get_xy() == xy).all()
```

Comment:
`pytest` cannot introspect into this. Therefore, I'd go with `assert_array_equal(polygon.get_xy(), xy)` (`from numpy.testing import assert_array_equal`). This gives better results in case of errors (cf. https://github.com/pytest-dev/pytest/issues/5347).

Reply:
Thanks for your suggestions. I've updated the PR accordingly.